### PR TITLE
Add alert to warn against posting nsec1 private keys

### PR DESCRIPTION
**Pull request by tyiu**
Simulator Screen Shot - iPhone 14 Pro - 2023-02-02 at 23 28 21
Simulator Screen Shot - iPhone 14 Pro - 2023-02-02 at 23 29 21

### CODE REVIEW

File: damus/Views/NoteContentView.swift [hunk 1]
```diff
@@ -211,7 +211,7 @@ struct NoteContentView: View {
                         if #available(iOS 16, *) {
                             noteLanguage = Locale.LanguageCode(stringLiteral: noteLanguage!).identifier(.alpha2)
                         } else {
-                            noteLanguage = Locale.canonicalLanguageIdentifier(from: noteLanguage!)
+                            noteLanguage = NSLocale(localeIdentifier: noteLanguage!).languageCode
```
Comment (tyiu) [inline damus/Views/NoteContentView.swift:214]: Unrelated bug fix. Translation logic wasn't working for < iOS 16

File: damus/Components/TranslateView.swift [hunk 2]
```diff
@@ -95,7 +95,7 @@ struct TranslateView: View {
                 if #available(iOS 16, *) {
                     noteLanguage = Locale.LanguageCode(stringLiteral: lang).identifier(.alpha2)
                 } else {
-                    noteLanguage = Locale.canonicalLanguageIdentifier(from: lang)
+                    noteLanguage = NSLocale(localeIdentifier: lang).languageCode
```
Comment (tyiu) [inline damus/Components/TranslateView.swift:98]: Unrelated bug fix. Translation logic wasn't working for < iOS 16

File: damus/Views/PostView.swift [hunk 3]
```diff
@@ -65,7 +66,16 @@ struct PostView: View {
 
                 if !is_post_empty {
                     Button(NSLocalizedString("Post", comment: "Button to post a note.")) {
-                        self.send_post()
+                        if #available(iOS 16.0, *) {
```
Comment (jb55) [inline damus/Views/PostView.swift:69]: this logic shouldn't be duplicated in two places

**Review by jb55 — CHANGES_REQUESTED** [review]

File: damus/Views/PostView.swift (hunk 3 above)
Comment (tyiu) [reply damus/Views/PostView.swift]: Refactored.

**Comment by jb55** [comment]
taking another look at this

**Comment by jb55** [comment]
perfect!
